src/common/utils.cpp

```diff
@@ -555,6 +558,147 @@ bool isGlobalKernelPrefix()
 }
 
 
+// Incapsulates Windows private namespace 
+class PrivateNamespace
+{
+public:
+	PrivateNamespace(MemoryPool& pool) :
+		m_hNamespace(NULL)
+	{
+		try
+		{
+			init();
+		}
+		catch (const Firebird::Exception& ex)
+		{
+			iscLogException("Error creating private namespace", ex);
+		}
+	}
+
+	~PrivateNamespace()
+	{
+		if (m_hNamespace != NULL)
+			ClosePrivateNamespace(m_hNamespace, 0);
+	}
+
+	// Add namespace prefix to the name, returns true on success.
+	bool addPrefix(char* name, size_t bufsize)
+	{
+		if (m_hNamespace == NULL)
+			return false;
+
+		if (strchr(name, '\\') != 0)
+			return false;
+
+		const size_t prefixLen = strlen(sPrivateNameSpace) + 1;
+		const size_t nameLen = strlen(name) + 1;
+		if (prefixLen + nameLen > bufsize)
+			return false;
+
+		memmove(name + prefixLen, name, nameLen + 1);
+		memcpy(name, sPrivateNameSpace, prefixLen - 1);
+		name[prefixLen - 1] = '\\';
+		return true;
+	}
+
+	bool isReady() const
+	{
+		return (m_hNamespace != NULL);
+	}
+
+private:
+	const char* sPrivateNameSpace = "FirebirdCommon";
+	const char* sBoundaryName = "FirebirdCommonBoundary";
+
+	void raiseError(const char* apiRoutine)
+	{
+		(Firebird::Arg::Gds(isc_sys_request) << apiRoutine << Firebird::Arg::OsError()).raise();
+	}
+
+	void init()
+	{
+		char sid[SECURITY_MAX_SID_SIZE];
```

Comment:
Can this be in the stack in unaligned location? Maybe some alignas should be used?

Here https://docs.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-sid I understand that AllocateAndInitializeSid would be needed.

Reply:
Here used [CreateWellKnownSid()](https://docs.microsoft.com/en-us/windows/win32/api/securitybaseapi/nf-securitybaseapi-createwellknownsid) and I see no requirements for sid alignment.
Also, looking for [usage samples](https://cpp.hotexamples.com/examples/-/-/CreateWellKnownSid/cpp-createwellknownsid-function-examples.html) one can see that its uses either [LocalAlloc()](https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-localalloc) or stack allocation. 
Also, I see no notes about alignment of allocated by LocalAlloc() memory.

Anyway, I'll add `alignas(SID)` to be sure.